### PR TITLE
feat(24.04): openjdk-17 headless slices

### PR DESCRIPTION
**Pull request by vpa1977**
# Proposed changes

Add openjdk-17 slices except AWT slice as it needs harfbuzz to be merged.

## Related issues/PRs

[harfbuzz](https://github.com/canonical/chisel-releases/pull/348)

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have tested my changes ([see how](https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md#7-test-your-slices-before-opening-a-pr))
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

File: slices/openjdk-17-jre-headless.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+package: openjdk-17-jre-headless
+
+essential:
+  - openjdk-17-jre-headless_copyright
+
+slices:
```
Comment (rebornplusplus) [inline slices/openjdk-17-jre-headless.yaml:6]: We should probably leave a one-liner comment on top of each slice describing what it does.

**Review by rebornplusplus — COMMENTED** [review]
Looks good in general. Just a few questions and suggestions below.

File: tests/spread/integration/openjdk-17-jre-headless/task.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  cp Main.java ${rootfs}/
+  cp ReadCertificate.java ${rootfs}/
+  cp certificate.pem ${rootfs}/
+  cd ${rootfs}
+  mkdir -p proc/self
+  for java in `find usr/lib/jvm -name java`; do
+    ln -sf /${java} proc/self/exe
+    chroot . ${java} --version
+    case ${SLICE} in
+      classescache)
```
[the remark this answers is not in the document]
Comment (vpa1977) [reply tests/spread/integration/openjdk-17-jre-headless/task.yaml:26]: Fixed, added back classescache slice.

File: slices/openjdk-17-jre-headless.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+  standard:
+    essential:
+      - openjdk-17-jre-headless_core
+      - openjdk-17-jre-headless_prefs
+      - openjdk-17-jre-headless_rmi
+      - openjdk-17-jre-headless_security
```
[the remark this answers is not in the document]
Comment (vpa1977) [reply slices/openjdk-17-jre-headless.yaml]: We may want to add debug/monitoring capabilities - default spring boot application requires them.

File: slices/openjdk-17-jre-headless.yaml [hunk 4]
```diff
@@ -0,0 +1,160 @@
+package: openjdk-17-jre-headless
```
[the remark this answers is not in the document]
Comment (vpa1977) [reply slices/openjdk-17-jre-headless.yaml:1]: `/usr/lib/jvm/java-17-openjdk-amd64/lib/classlist                                                 openjdk-17-jre-headless` belong to classescache slice, .jsa files are generated with `java -Xdump`

`/usr/lib/jvm/java-17-openjdk-amd64/lib/libjsvml.so                                               openjdk-17-jre-headless` - incubating feature, vector API, omitted
`/usr/lib/jvm/java-17-openjdk-amd64/lib/libsyslookup.so                                           openjdk-17-jre-headless` -  incubating feature, Foreign Function API, omitted. 

Rest are awt slice which is not included in the MR.

**Comment by vpa1977** [comment]
Needs  pipeline retry  - spread failing with "cannot fetch from archive"

**Review by rebornplusplus — APPROVED** [review]
Looks good to me, thank you!

I have re-triggered the failing pipeline. It's passing now.

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
again the same comments as in https://github.com/canonical/chisel-releases/pull/313 and #349

**Comment by vpa1977** [comment]
Note: dropped jvm.cfg-default https://github.com/canonical/chisel-releases/pull/363

**Review by rebornplusplus — APPROVED** [review]
Looks great, thank you very much!

**Review by cjdcordeiro — APPROVED** [review]

**Pull request merged**